ert_shared/ensemble_evaluator/entity/prefect_ensamble.py

```diff
@@ -0,0 +1,287 @@
+                                 real_id=self._iens,
+                                 step_id=self._step_id,
+                                 stage_id=self._stage_id
+                                 )
+            c.send(to_json(event))
+        return exec_metadata
+
+
+def _get_event_type(state, fm_type):
+    if isinstance(state, prefect.engine.state.Pending):
+        return f"com.equinor.ert.forward_model_{fm_type}.start"
+    elif isinstance(state, prefect.engine.state.Running):
+        return f"com.equinor.ert.forward_model_{fm_type}.running"
+    elif isinstance(state, prefect.engine.state.Success):
+        return f"com.equinor.ert.forward_model_{fm_type}.success"
+    elif isinstance(state, prefect.engine.state.Failed):
+        return f"com.equinor.ert.forward_model_{fm_type}.failure"
+    else:
+        return f"com.equinor.ert.forward_model_{fm_type}.unknown"
+
+
+def _build_source(fm_type, real_id, stage_id=None,  step_id=None, job_id=None):
+    source_map = {
+        "stage": f"/ert/ee/0/real/{real_id}/stage/{stage_id}",
+        "step": f"/ert/ee/0/real/{real_id}/stage/{stage_id}/step/{step_id}",
+        "job": f"/ert/ee/0/real/{real_id}/stage/{stage_id}/step/{step_id}/job/{job_id}",
+    }
+    return source_map.get(fm_type, f"/ert/ee/0")
+
+
+def _cloud_event(event_type, fm_type, real_id, stage_id=None, step_id=None, job_id=None, data=None):
+    if data is None:
+        data = {}
+    if fm_type == "stage":
+        data.update({"queue_event_type": ""})
+
+    return CloudEvent(
+        {
+            "type": event_type,
+            "source": _build_source(fm_type, real_id, stage_id, step_id, job_id),
+            "datacontenttype": "application/json",
+        },
+        data
+    )
+
+
+def gen_coef(parameters, real):
+    data = {}
+    paths = {}
+    for iens in range(real):
+        for name, elements in parameters.items():
+            for element in elements:
+                start, end = element["args"]
+                data[element["name"]] = uniform(start, end)
+        os.makedirs(f"coeff/{iens}", exist_ok=True)
+        file_name = f"coeff/{iens}/coeffs.json"
+        with open(file_name, "w") as f:
+            json.dump(data, f)
+        paths[iens] = os.path.abspath(file_name)
+    return paths
+
+
+class PrefectEnsemble(_Ensemble):
+    def __init__(self, config):
+        self.config = config
+        self._reals = self._get_reals()
+        super().__init__(self._reals, metadata={"iter": 0})
+
+    def _get_reals(self):
+        reals = []
+        for iens in range(self.config['realizations']):
+            stages = []
+            for stage in self.config['stages']:
+                steps = []
+                stage_id = uuid.uuid4()
+                for step in stage["steps"]:
+                    jobs = []
+                    for job in step["jobs"]:
+                        job_id = uuid.uuid4()
+                        jobs.append(_ScriptJob(id_=str(job_id),
```

Comment:
Could you create a new type of job? The `ee-legacy` branch turns the `ScriptJob` into a `LegacyJob` that does not have `executable` and `args`.

I thought that, in our discussions, you mentioned that you will never use the `script/unix` type job?

Reply:
Why do we need to distinguish any type of job from another at this point? Why does the evaluator/UI need to know any information about what kind of job it is, or what it is doing?

Reply:
Because we do this today… And we want to keep doing it?

Reply:
I think there is something fundamental I am not understanding. What code in the end distinguishes between these types of jobs? And who needs information about what it concretely represents, such as executable?

Reply:
In the near future, I assume we will attempt to take ownership—in the `EnsembleEvaluator`—of the task of writing a `jobs.json`. My reasoning is that the legacy evaluator will survive the move to cloud, and that the shared disk assumption in the current code tasked with creating `jobs.json`, will not (should not) work in the cloud.

In order to do this, the evaluator (or the legacy ensemble) needs e.g. `executable`.

Reply:
I think you're right that it doesn't really matter whether it's a `LegacyJob` or a `WhateverJob`, as long as they share an interface.

But I feel uneasy making decisions about who needs what when we're supposed to make a general-purpose ensemble evaluator.